src/GitHubExtension/DataManager/GitHubDataManager.cs

```diff
@@ -703,6 +725,41 @@ private async Task UpdateIssuesAsync(Repository repository, Octokit.GitHubClient
         Issue.DeleteLastObservedBefore(DataStore, repository.Id, DateTime.UtcNow - LastObservedDeleteSpan);
     }
 
+    // Internal method to update releases. Assumes Repository has already been populated and created.
+    // DataStore transaction is assumed to be wrapped around this in the public method.
+    private async Task UpdateReleasesAsync(Repository repository, Octokit.GitHubClient? client = null, RequestOptions? options = null)
+    {
+        options ??= RequestOptions.RequestOptionsDefault();
+
+        // Limit the number of fetched releases.
+        options.ApiOptions.PageCount = 1;
+        options.ApiOptions.PageSize = 50;
+
+        client ??= await GitHubClientProvider.Instance.GetClientForLoggedInDeveloper(true);
+        Log.Logger()?.ReportInfo(Name, $"Updating releases for: {repository.FullName}");
+
+        var releasesResult = await client.Repository.Release.GetAll(repository.InternalId, options.ApiOptions);
+        if (releasesResult == null)
+        {
+            Log.Logger()?.ReportDebug($"No releases found.");
+            return;
+        }
+
+        Log.Logger()?.ReportDebug(Name, $"Results contain {releasesResult.Count} releases.");
+        foreach (var release in releasesResult)
+        {
+            if (release.Draft)
+            {
+                continue;
+            }
+
+            _ = Release.GetOrCreateByOctokitRelease(DataStore, release, repository.Id);
+        }
+
+        // Remove releases from this repository that were not observed recently.
+        Release.DeleteLastObservedBefore(DataStore, repository.Id, DateTime.UtcNow - LastObservedDeleteSpan);

```

Comment:
This will catch releases being tracked by a widget and remove older ones as newer ones come in (for example, if a new release is encountered and we are over page size 10, then the oldest will be removed) for this repository only.

This will not catch situations such as removing a repository widget and no longer tracking the releases at all, in which case they'd stay in the datastore forever. The Github widget has this problem in a few other places that need to be fixed but moving forward we do want to ensure the data is cleaned up if the widget is removed. PruneObsoleteData() does this, so this data object needs a delete to catch unobserved data because there is no widget for that repository anymore. That one could have a longer last observed delete span (perhaps a few days) defined in the data object.